rpc/serializers.py

```diff
@@ -350,13 +351,64 @@ def get_target_draft_name(self, obj: RpcRelatedDocument) -> str:
         return obj.target_rfctobe.draft.name
 
 
-class CreateRpcRelatedDocumentSerializer(serializers.ModelSerializer):
+class CreateRpcRelatedDocumentSerializer(RpcRelatedDocumentSerializer):
     """Serializer for creating a related document for an RfcToBe"""
 
-    # todo reconcile with refactored RpcRelatedDocumentSerializer
+    target_draft_name = serializers.CharField(write_only=True, required=False)
+    target_draft_name_output = serializers.SerializerMethodField(read_only=True)
+
     class Meta:
         model = RpcRelatedDocument
-        fields = ["id", "relationship", "source", "target_document", "target_rfctobe"]
+        fields = [
+            "id",
+            "relationship",
+            "source",
+            "target_draft_name",
+            "target_draft_name_output",
+        ]
+
+    def get_target_draft_name_output(self, obj):
+        if obj.target_document is not None:
+            return obj.target_document.name
+        if obj.target_rfctobe is not None:
+            return obj.target_rfctobe.draft.name
+        return None
+
+    def to_representation(self, instance):
+        ret = super().to_representation(instance)
+        ret["target_draft_name"] = ret.pop("target_draft_name_output", None)
+        return ret
+
+    def create(self, validated_data):
+        target_draft_name = validated_data.pop("target_draft_name")
+
+        source = validated_data["source"]
+
+        target_document = Document.objects.filter(name=target_draft_name).first()
```

Comment:
You need to check for an RfcToBe first. If there is an RfcToBe, there will always be a Document, so this will always link to a Document or fail validation.